src/core/textrenderer/qgstextbackgroundsettings.cpp

```diff
@@ -18,13 +18,26 @@
 #include "qgsvectorlayer.h"
 #include "qgspallabeling.h"
 #include "qgssymbollayerutils.h"
+#include "qgsfillsymbollayer.h"
 #include "qgspainting.h"
 #include "qgstextrendererutils.h"
 #include "qgspainteffectregistry.h"
 
 QgsTextBackgroundSettings::QgsTextBackgroundSettings()
 {
   d = new QgsTextBackgroundSettingsPrivate();
+
+  // Create a default fill symbol to preserve API promise until QGIS 4.0
+  QgsSimpleFillSymbolLayer *fill = new QgsSimpleFillSymbolLayer( d->fillColor, Qt::SolidPattern, d->strokeColor );
```

Comment:
I think that a unique_ptr should be used in this case.

Reply:
@pblottiere , I don't think it's necessary, the ownership is all forwarded to the fill symbol, and ultimately to setFillSymbol( ... ) a few lines below.

Reply:
Personally, I always prefer to use `unique_ptr` instead of `new` because I think it's more robust to  future modifications. Moreover, it'd probably more homogeneous with existing code. For example:

https://github.com/qgis/QGIS/blob/e66e305a40e948806ba40dd933a5df35759e3189/src/core/layout/qgslayoutitemscalebar.cpp#L92-L101

But +1 to ignore my comment if you think it's not necessary :).

Reply:
@pblottiere , I'll stick to a good old pointer here, but your point's taken. On that note, can you give this PR an approval? :) 

BTW, your recent inflow of energy review PRs is greatly appreciated.

Reply:
> BTW, your recent inflow of energy review PRs is greatly appreciated.

Thanks @nirvn. I'm probably not the most knowledgeable person to review core/ui QGIS source code, but at least I'll do my best to take a look on coding rules and guidelines :).